sdk/iotcentral/Azure.ResourceManager.IotCentral/tests/Tests/IotCentralExtensionsTests.cs

```diff
@@ -0,0 +1,152 @@
+            await iotCentralApp.DeleteAsync(WaitUntil.Completed, CancellationToken.None);
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task IotCentralListSubscriptionApplicationsTest()
+        {
+            // Get IoT Central apps collection for resource group.
+            var subscription = Client.GetSubscriptionResource(new ResourceIdentifier($"/subscriptions/{SessionEnvironment.SubscriptionId}"));
+            var rg = await CreateResourceGroupAsync(subscription, "sdk-test-rg-");
+
+            var appsCollection = rg.GetIotCentralApps();
+
+            // Create IoT Central applications.
+            var appNames = new List<string>() { GetRandomTestName(), GetRandomTestName(), GetRandomTestName() };
+            foreach (var appName in appNames)
+            {
+                var iotCentralAppData = new IotCentralAppData(AzureLocation.WestUS, new AppSkuInfo(AppSku.ST0))
+                {
+                    DisplayName = appName,
+                    Subdomain = appName,
+                };
+                await appsCollection.CreateOrUpdateAsync(WaitUntil.Completed, appName, iotCentralAppData, CancellationToken.None);
+            }
+
+            // Get and delete IoT Central apps.
+            await foreach (var subApp in subscription.GetIotCentralAppsAsync(CancellationToken.None))
+            {
+                if (appNames.Contains(subApp.Data.Name))
+                {
+                    appNames.Remove(subApp.Data.Name);
+                    await subApp.DeleteAsync(WaitUntil.Completed, CancellationToken.None);
+                }
+            }
+
+            // Assert all apps created were found and deleted.
+            Assert.IsFalse(appNames.Any());
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task IotCentralListResourceGroupApplicationsTest()
+        {
+            // Get IoT Central apps collection for resource group.
+            var subscription = Client.GetSubscriptionResource(new ResourceIdentifier($"/subscriptions/{SessionEnvironment.SubscriptionId}"));
+            var rg = await CreateResourceGroupAsync(subscription, "sdk-test-rg-");
+
+            var appsCollection = rg.GetIotCentralApps();
+
+            // Create IoT Central applications.
+            var appNames = new List<string>() { GetRandomTestName(), GetRandomTestName(), GetRandomTestName() };
+            foreach (var appName in appNames)
+            {
+                var iotCentralAppData = new IotCentralAppData(AzureLocation.WestUS, new AppSkuInfo(AppSku.ST0))
+                {
+                    DisplayName = appName,
+                    Subdomain = appName,
+                };
+                await appsCollection.CreateOrUpdateAsync(WaitUntil.Completed, appName, iotCentralAppData, CancellationToken.None);
+            }
+
+            // Get and delete IoT Central apps.
+            await foreach (var rgApp in rg.GetIotCentralApps())
```

Comment:
Nit: can we directly call the `GetAll` method?
```suggestion
            await foreach (var rgApp in appsCollection.GetAllAsync())
```

Reply:
this test is meant to check whether that rg.GetIotCentralApps() calls returns all the created applications.

Reply:
Yeah, I understand your intention and I'm ok with that. I have this suggestion because the `IotCentralAppCollection` class is derived from [IAsyncEnumerable](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/iotcentral/Azure.ResourceManager.IotCentral/src/Generated/IotCentralAppCollection.cs#L28), so this call will implicitly call the [GetAsyncEnumerator](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/iotcentral/Azure.ResourceManager.IotCentral/src/Generated/IotCentralAppCollection.cs#L329) method, in which the `GetAllAsync()` method is called to return all the created apps in this resource group. So I'd like to directly use the `GetAllAsync()` method to make the code more intuitive.

Reply:
i see, updated!